Make plugin load async instead of upon style resolution

rniwa opened the pull request:
#### f4b10fa093ec5898db23e16ab60ee81791098b4a
<pre>
Make plugin load async instead of upon style resolution
<a href="https://bugs.webkit.org/show_bug.cgi?id=242789">https://bugs.webkit.org/show_bug.cgi?id=242789</a>

Reviewed by Geoffrey Garen.

This patch makes plugin load fully asynchronous instead of loading at the end of style resolution.

Some tests which load HTML content in object / embed contents are updated to wait for rAF then 0s timer.
This is needed to accommodate this new behavior in these tests.

* LayoutTests/editing/editability/ignored-content.html:
* LayoutTests/editing/execCommand/outdent-paragraph-crash.html: Made this test more robust.
* LayoutTests/fast/frames/sandboxed-iframe-plugins.html:
* LayoutTests/http/tests/misc/embed-image-load-outlives-gc-without-crashing.html:
* LayoutTests/webarchive/loading/object-expected.txt:

* Source/WebCore/html/HTMLPlugInImageElement.cpp:
(WebCore::HTMLPlugInImageElement::scheduleUpdateForAfterStyleResolution):

Canonical link: <a href="https://commits.webkit.org/252541@main">https://commits.webkit.org/252541@main</a>
</pre>

geoffreygaren reviewed (approved):
r=me

Is this a compatibility change? If so, would be good to add a test. Maybe an <object> with a javascript: URL could verify whether it executes sync or async.

rniwa committed:
Make plugin load async instead of upon style resolution
https://bugs.webkit.org/show_bug.cgi?id=242789

Reviewed by Geoffrey Garen.

This patch makes plugin load fully asynchronous instead of loading at the end of style resolution.

Some tests which load HTML content in object / embed contents are updated to wait for rAF then 0s timer.
This is needed to accommodate this new behavior in these tests.

* LayoutTests/editing/editability/ignored-content.html:
* LayoutTests/editing/execCommand/outdent-paragraph-crash.html: Made this test more robust.
* LayoutTests/fast/frames/sandboxed-iframe-plugins.html:
* LayoutTests/http/tests/misc/embed-image-load-outlives-gc-without-crashing.html:
* LayoutTests/webarchive/loading/object-expected.txt:

* Source/WebCore/html/HTMLPlugInImageElement.cpp:
(WebCore::HTMLPlugInImageElement::scheduleUpdateForAfterStyleResolution):

Canonical link: https://commits.webkit.org/252541@main
